npm/test/clientfileprocessing.test.ts

```diff
@@ -172,7 +172,7 @@ test("client file metadata successfully uploaded", async () => {
   ).resolves.not.toThrow()
 })
 
```

Comment:
I would refactor the test slightly so you have tests as follows to have the following tests:
- "metadataProgressBarCallback function updates the progress bar with the percentage processed"
- "metadataProgressBarCallback function updates the progress bar to a maximum of 50 percent"
-- with this test set the percent processed to 100% to make sure the @value displays only 50
- "metadataProgress function does not change the HTML state if no progress bar present"
- "s3ProgressBarCallback function updates progress bar from a minimum of 50 percent"
-- with this test set the percent processed to 0% and test the @value displays 50
-  "s3ProgressBarCallback function updates the progress bar with the percentage processed"
- "s3ProgressBarCallback function updates the progress bar to a maximum of 100 percent"

This set of tests should then cover the different behaviours of the the two progress bar callback functions. For the s3 function tests you can write a separate check function to make sure the @value is set correctly